packages/tenant-process/src/services/tenantService.ts

```diff
@@ -878,4 +967,33 @@ function reassignVerifiedAttribute(
   );
 }
 
+async function revokeCertifiedAttribute(
+  tenant: Tenant,
+  readModelService: ReadModelService,
+  attributeId: AttributeId
+): Promise<Tenant> {
+  const updatedTenant: Tenant = {
+    ...tenant,
+    updatedAt: new Date(),
+    attributes: tenant.attributes.map((attr) =>
+      attr.id === attributeId
+        ? {
+            ...attr,
+            revocationTimestamp: new Date(),
+          }
+        : attr
+    ),
+  };
+
+  const updatedKind = await getTenantKindLoadingCertifiedAttributes(
+    readModelService,
+    updatedTenant.attributes,
+    updatedTenant.externalId
+  );
+
+  return updatedTenant.kind === updatedKind
+    ? updatedTenant
+    : { ...updatedTenant, kind: updatedKind };
```

Comment:
this is done [here](https://github.com/pagopa/interop-be-monorepo/pull/845/files#diff-c52f69aae83f1d150aa67f380d299976f1a67b9cb76738048267062cdb848f3eR670)

Reply:
I preferred to remove the `kind` logic from this function and insert it into the main function, as we do with the others